File: java/Graph/graph.java

class graph {
    int gp[][] = new int[10][10];
    int vertex;
    graph(int v) {
        this.vertex = v;
        for(int i = 0; i < v; i++) {
            for(int j = 0; j < v; j++) {
                this.gp[i][j] = 0;
            }
        }
    }

    public void printGraph() {
        for(int i = 0; i < this.vertex; i++) {
            for(int j = 0; j < this.vertex; j++) {
                System.out.print(this.gp[i][j] + " ");
            }
            System.out.println();
        }
    }

    public void add(int s, int d) {
        if(s < 1 || s >= this.vertex || d < 1 || d >= this.vertex) {
            System.out.println("Not a valid edge!");
        } 
        // assume bidirected graph
        this.gp[s-1][d-1] = 1;
        this.gp[d-1][s-1] = 1;

    }
 

}


public class graph {
    public static void main(String args[]) {
        graph g1 = new graph(3);
        g1.add(1, 2);
        g1.printGraph();
        System.out.println("heleo");
    }
}

File: cpp/Graph/graph.cpp
#include <iostream>
#include <vector>
#include <list>
#include <queue>

// A class to represent a graph.
class Graph {
private:
    int numVertices;
    // Adjacency list: array of lists of integers
    std::vector<std::list<int>> adj;

public:
    // Constructor
    Graph(int vertices) {
        this->numVertices = vertices;
        adj.resize(vertices);
    }

    // Function to add an edge to an undirected graph
    void addEdge(int src, int dest) {
        // Add an edge from src to dest.
        adj[src].push_back(dest);
        // Since the graph is undirected, add an edge from dest to src also
        adj[dest].push_back(src);
    }

    // BFS traversal from a given source vertex
    void BFS(int startVertex) {
        // Create a visited array and initialize all vertices as not visited
        std::vector<bool> visited(numVertices, false);

        // Create a queue for BFS
        std::queue<int> queue;

        // Mark the current node as visited and enqueue it
        visited[startVertex] = true;
        queue.push(startVertex);

        std::cout << "BFS starting from vertex " << startVertex << ": ";

        while (!queue.empty()) {
            // Dequeue a vertex from queue and print it
            int currentVertex = queue.front();
            std::cout << currentVertex << " ";
            queue.pop();

            // Get all adjacent vertices of the dequeued vertex currentVertex.
            // If an adjacent has not been visited, then mark it visited and enqueue it.
            for (int adjacent : adj[currentVertex]) {
                if (!visited[adjacent]) {
                    visited[adjacent] = true;
                    queue.push(adjacent);
                }
            }
        }
        std::cout << std::endl;
    }
};

int main() {
    // Create a graph with 7 vertices, similar to the one in your Java BFS example
    Graph g(7);

    // Add edges to the graph
    g.addEdge(0, 1);
    g.addEdge(0, 2);
    g.addEdge(1, 3);
    g.addEdge(2, 4);
    g.addEdge(3, 4);
    g.addEdge(3, 5);
    g.addEdge(4, 5);
    g.addEdge(5, 6);

    // Perform BFS traversal starting from vertex 0
    g.BFS(0);

    return 0;
}